app/src/models/branch.ts

```diff
@@ -45,12 +45,14 @@ export class Branch {
    * @param upstream The remote-prefixed upstream name. E.g., `origin/main`.
    * @param tip Basic information (sha and author) of the latest commit on the branch.
    * @param type The type of branch, e.g., local or remote.
+   * @param ref The connical ref of the branch
    */
   public constructor(
     public readonly name: string,
     public readonly upstream: string | null,
     public readonly tip: IBranchTip,
-    public readonly type: BranchType
+    public readonly type: BranchType,
+    public readonly ref: string
```

Comment:
Per my suggestion above:
> **If** we accept it's valid not having a proper `ref`, I'd say we state that clearly by making `ref` be of type `string | null`.
```suggestion
    public readonly ref: string | null
```